includes/WikiPage.php

```diff
@@ -1297,6 +1297,12 @@ public function doEdit( $text, $summary, $flags = 0, $baseRevId = false, $user =
 
 		$flags = $this->checkFlags( $flags );
 
+		$links = null;
+		Hooks::run( 'ArticleGetFileLinks', [ $this->mTitle->getArticleID(), &$links ] );
```

Comment:
I don't think this flow is a good way to handle this. Having a hook populate a variable to be used by a later hook is a very complex flow. I assume this is because the data is gone to generate the links list by this stage. I think instead, we should either purge the cache on the earlier `ArticleDelete` hook (it's probably OK if a few purges happen when the article isn't successfully deleted), or at worst, add a hook in a similar place to `ArticleGetFileLinks` that does the purge there, but I think `ArticleDelete` would be fine.

And why is `ArticleGetFileLinks` used twice?

Reply:
We have two cases: a new link is added(it seems right now i doesn't clear all the data from cache, so we need to run a full purge on all the links ) and an old link is deleted (that's why I've added the hook to populate links before changes happen). Purging cache before the data is deleted from the db might cause caching of an old redirect (sometimes taskPublisher->doUpdate() fires before action ends and instead of one request to celery we have few more). I'll try to check it tomorrow

Reply:
It seems I must have forced sending data to celery while I was debuging. For now the requests are sent after the page loop finished. thanks :)